src/domain/signupGroup/editButtonPanel/EditSignupGroupButtonPanel.tsx

```diff
@@ -0,0 +1,129 @@
+import { IconPen } from 'hds-react';
+import React, { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import ButtonPanel from '../../../common/components/buttonPanel/ButtonPanel';
+import buttonPanelStyles from '../../../common/components/buttonPanel/buttonPanel.module.scss';
+import LoadingButton from '../../../common/components/loadingButton/LoadingButton';
+import { MenuItemOptionProps } from '../../../common/components/menuDropdown/types';
+import { ROUTES } from '../../../constants';
+import {
+  RegistrationFieldsFragment,
+  SignupGroupFieldsFragment,
+} from '../../../generated/graphql';
+import useGoBack from '../../../hooks/useGoBack';
+import getValue from '../../../utils/getValue';
+import skipFalsyType from '../../../utils/skipFalsyType';
+import { useAuth } from '../../auth/hooks/useAuth';
+import { ENROLMENT_ACTIONS } from '../../enrolment/constants';
+import { getEditButtonProps } from '../../enrolment/utils';
+import { EnrolmentsLocationState } from '../../enrolments/types';
+import useOrganizationAncestors from '../../organization/hooks/useOrganizationAncestors';
+import useUser from '../../user/hooks/useUser';
+import { SIGNUP_GROUP_ACTIONS } from '../constants';
+import { getSignupGroupActionButtonProps } from '../permissions';
+import { getResponsiblePerson } from '../utils';
+import styles from './editButtonPanel.module.scss';
+
+export interface EditSignupGroupButtonPanelProps {
+  onSendMessage: () => void;
+  onUpdate: () => void;
+  registration: RegistrationFieldsFragment;
+  saving: SIGNUP_GROUP_ACTIONS | null;
+  signupGroup: SignupGroupFieldsFragment;
+}
+
+const EditSignupGroupButtonPanel: React.FC<EditSignupGroupButtonPanelProps> = ({
+  onSendMessage,
+  onUpdate,
+  registration,
+  saving,
+  signupGroup,
+}) => {
+  const { t } = useTranslation();
+  const { isAuthenticated: authenticated } = useAuth();
+  const { user } = useUser();
+  const publisher = getValue(registration.publisher, '');
+  const { organizationAncestors } = useOrganizationAncestors(publisher);
+
+  const responsiblePerson = useMemo(
+    () => getResponsiblePerson(signupGroup),
+    [signupGroup]
+  );
+
+  const goBack = useGoBack<EnrolmentsLocationState>({
+    defaultReturnPath: ROUTES.REGISTRATION_ENROLMENTS.replace(
+      ':registrationId',
+      getValue(registration.id, '')
+    ),
+    state: { enrolmentId: responsiblePerson?.id },
+  });
+
+  const getSingupActionItemProps = ({
```

Comment:
Small typo here ☝️

Reply:
Great catch from such huge amount of lines